crates/biome_js_analyze/tests/specs/nursery/noFlatMapIdentity/invalid.js

```diff
@@ -1,2 +1,7 @@
+array.flatMap(function f(arr) { return arr });
+
+array.flatMap(function (arr) { return arr });
+
 array.flatMap((arr) => arr)
+
 array.flatMap((arr) => {return arr})
```

Comment:
Sorry I wasn't being clear. IMO the third and fourth test cases in https://github.com/biomejs/biome/pull/2324/commits/f9937487ae45dd78c42882d4799757cbe18e42d8#r1555418726 should also be added to ensure this rule is able to deal with arrow function callbacks without parentheses around the arguments:

We now have:

```js
array.flatMap((arr) => arr)

array.flatMap((arr) => {return arr})
```

I'd like to also have

```js
array.flatMap(arr => arr)

array.flatMap(arr => {return arr})
```

Same for the valid case.